diff --git a/src/Plugin/facets/widget/DateWidget.php b/src/Plugin/facets/widget/DateWidget.php
index 1267c7f..b523e03 100644
--- a/src/Plugin/facets/widget/DateWidget.php
+++ b/src/Plugin/facets/widget/DateWidget.php
@@ -136,6 +136,10 @@ class DateWidget extends ListPagesWidgetBase {
 
     foreach ($value_keys as $key) {
       $value = $form_state->getValue($key);
+      if (!$value) {
+        continue;
+      }
+
       if (!$value instanceof DrupalDateTime) {
         $value = new DrupalDateTime($value);
       }
@@ -143,6 +147,11 @@ class DateWidget extends ListPagesWidgetBase {
       $values[] = $value->format(\DateTimeInterface::ATOM);
     }
 
+    if (count($values) === 1) {
+      // If we only have the operator, it means no dates have been specified.
+      return [];
+    }
+
     return [implode('|', $values)];
   }
 
